Wangscape/Reseedable.cpp

```diff
@@ -0,0 +1,285 @@
+{
+    std::shared_ptr<noise::module::Blend> blend_p = std::make_shared<noise::module::Blend>();
+    blend_p->SetSourceModule(0, *source_a.module);
+    blend_p->SetSourceModule(1, *source_b.module);
+    blend_p->SetControlModule(*module);
+
+    std::shared_ptr<ModuleGroup> mg = std::make_shared<ModuleGroup>();
+    mg->insert("control", *this);
+    mg->insert("source0", source_a);
+    mg->insert("source1", source_b);
+    mg->insert("output", makeReseedable(blend_p));
+    return makeReseedable(mg);
+}
+
+Reseedable Reseedable::select(Reseedable& source_a, Reseedable& source_b)
+{
+    std::shared_ptr<noise::module::Select> select_p = std::make_shared<noise::module::Select>();
+    select_p->SetSourceModule(0, *source_a.module);
+    select_p->SetSourceModule(1, *source_b.module);
+    select_p->SetControlModule(*module);
+
+    std::shared_ptr<ModuleGroup> mg = std::make_shared<ModuleGroup>();
+    mg->insert("control", *this);
+    mg->insert("source0", source_a);
+    mg->insert("source1", source_b);
+    mg->insert("output", makeReseedable(select_p));
+    return makeReseedable(mg);
+}
+
+Reseedable Reseedable::rotatePoint(double x_angle, double y_angle, double z_angle)
+{
+    std::shared_ptr<noise::module::RotatePoint> rotate_point_p = std::make_shared<noise::module::RotatePoint>();
+    rotate_point_p->SetSourceModule(0, *module);
+    rotate_point_p->SetXAngle(x_angle);
+    rotate_point_p->SetYAngle(y_angle);
+    rotate_point_p->SetZAngle(z_angle);
+
+    std::shared_ptr<ModuleGroup> mg = std::make_shared<ModuleGroup>();
+    mg->insert("source", *this);
+    mg->insert("output", makeReseedable(rotate_point_p));
+    return makeReseedable(mg);
+}
+
+Reseedable Reseedable::scalePoint(double x_scale, double y_scale, double z_scale)
+{
+    std::shared_ptr<noise::module::ScalePoint> scale_point_p = std::make_shared<noise::module::ScalePoint>();
+    scale_point_p->SetSourceModule(0, *module);
+    scale_point_p->SetXScale(x_scale);
+    scale_point_p->SetYScale(y_scale);
+    scale_point_p->SetZScale(z_scale);
+
+    std::shared_ptr<ModuleGroup> mg = std::make_shared<ModuleGroup>();
+    mg->insert("source", *this);
+    mg->insert("output", makeReseedable(scale_point_p));
+    return makeReseedable(mg);
+}
+
+Reseedable Reseedable::translatePoint(double x_displace, double y_displace, double z_displace)
+{
+    std::shared_ptr<noise::module::TranslatePoint> translate_point_p = std::make_shared<noise::module::TranslatePoint>();
+    translate_point_p->SetSourceModule(0, *module);
+    translate_point_p->SetXTranslation(x_displace);
+    translate_point_p->SetYTranslation(y_displace);
+    translate_point_p->SetZTranslation(z_displace);
+
+    std::shared_ptr<ModuleGroup> mg = std::make_shared<ModuleGroup>();
+    mg->insert("source", *this);
+    mg->insert("output", makeReseedable(translate_point_p));
+    return makeReseedable(mg);
+}
+
+Reseedable Reseedable::turbulence()
+{
+    throw;
```

Comment:
Better to just implement the function :) but in future I'll use `std::terminate` or `throw` a real exception when marking an unimplemented function, rather than `throw` nothing.